File: push/src/main/java/com/xiaomi/push/service/XMPushServiceInboundPacketListener.java
package com.xiaomi.push.service;

import com.xiaomi.slim.Blob;
import com.xiaomi.smack.PacketListener;
import com.xiaomi.smack.packet.Packet;

final class XMPushServiceInboundPacketListener implements PacketListener {
    private final XMPushService service;

    XMPushServiceInboundPacketListener(XMPushService xMPushService) {
        this.service = xMPushService;
    }

    @Override
    public void process(Blob blob) {
        this.service.executeJob(new BlobReceiveJob(this.service, blob));
    }

    @Override
    public void processPacket(Packet packet) {
        this.service.executeJob(new PacketReceiveJob(this.service, packet));
    }
}

File: legacy/src/main/java/com/xiaomi/push/service/XMPushServiceInboundPacketListener.kt
package com.xiaomi.push.service
import io.github.magisk317.mipush.protocol.model.*

import com.xiaomi.slim.Blob
import com.xiaomi.smack.PacketListener
import com.xiaomi.smack.packet.Packet

class XMPushServiceInboundPacketListener(
    private val service: XMPushService,
) : PacketListener {
    override fun process(blob: Blob) {
        service.executeJob(BlobReceiveJob(service, blob))
    }

    override fun processPacket(packet: Packet) {
        service.executeJob(PacketReceiveJob(service, packet))
    }
}
